src/chain.rs

```diff
@@ -69,7 +72,7 @@ pub trait PevmChain: Debug {
         spec_id: SpecId,
         txs: &BlockTransactions<Self::Transaction>,
         tx_results: &[PevmTxExecutionResult],
-    ) -> B256;
+    ) -> Result<Option<B256>, Self::CalculateReceiptRootError>;
```

Comment:
This is more complicated than needed. If we promote it to `Result` then the `None` should be an `Error`, like `MissingPrevStateRoot`, `Unsupported`, or even `()`.

Reply:
ok, let me propose another version that use `Result<B256, CalculateReceiptRootError>`